src/control/createIndex.ts

```diff
@@ -25,6 +28,21 @@ export type CreateIndexOptions = {
   /** The dimension of the index. Must be a positive integer. The dimension of your index should match the output dimension of your embedding model. For example, if you are using a model that outputs 128-dimensional vectors, you should set the dimension to 128. */
   dimension: number;
 
+  /**
+   * The region where you would like your index to be created
+   */
+  region: string;
+
+  /**
+   * The public cloud where you would like your index hosted
+   */
+  cloud: 'gcp' | 'aws' | 'azure';
```

Comment:
This is aligned with the `CreateRequestCloudEnum` in the generated code: https://github.com/pinecone-io/pinecone-ts-client/pull/153/files#diff-678ee0ec5845c7d940569446c8f0a4210f8e23c95500729b0788439fd9759749R124

Maybe it makes more sense to just export and reuse that directly?

Reply:
It would be nice to adjust the spec to somehow export these directly in the compiled code. But this seems okay for now.